### DISCUSSION TITLE
Should we have separate discussions per-repo?

### DISCUSSION THREAD

**Discussion by austinlparker** [open-telemetry/community#583] [Community Discussions]
Kinda a general question - I'm interested in what other people think. In my mind there's pros and cons:

In favor of per-repo discussions:
- Discussions 'stay close' to their associated topic, which makes it easier for maintainers to be aware of discussions that are happening.
- Theoretically it becomes easier for end-users to find the right place to ask their questions, just go to the language you care about.

Against:
- There's an awful lot of cross-cutting concerns in OTel. If you're posting looking for help for an issue with, say, interaction between a java and a go service, where does that go? Java discussion or Go discussion? Are the right people going to be watching that discussion?
- Community members trying to provide support will need to look at many different repositories rather than one, although I suppose this is status quo, so maybe it's not a huge mark against it.

Other points:
- Is there going to be a moderation burden for maintainers? StackOverflow, for its faults, has moderation/administrators who can deal with some stuff. It may not turn out to be a problem, but it's good to think about.
- If we do have a single repo for discussions, is this the right repo, or should we make a stub repo that's only purpose is to hold discussions?

**Comment by Oberon00** [comment]
I think I would prefer an organization-wide discussion, and this repository seems fine, given the name. If we notice that there is too much traffic on the boards here, we can still decide to enable per-repository discussions. On the other hand, if we find we need a channel/subforum/category per repository in this main forum, I'd rather have per-repository discussions.

**Reply by Oberon00** [reply]
Oh, one thing I just realized is that watching would be better with per-repo discussions. It seems with a global discussion board, I can either watch it or not, and have no further filtering mechanism.

**Reply by austinlparker** [reply]
Yeah, it also seems like email notifications just lump everything together (so a reply to a discussion thread is the same as a reply to an issue, etc.)

**Reply by Oberon00** [reply]
You can watch only the discussion of a repository though, under "(Un)Watch -> Custom"

**Reply by austinlparker** [reply]
Oh, hm, that's not bad.

**Reply by dyladan** [reply]
I don't see a reason we can't have both. Discussions on sig repositories could be a good way to make language-specific decisions while more important/global decisions can be had on a community or spec repo.

**Comment by tigrannajaryan** [comment]
Is this a reversible decision, i.e. if we go with one centralized repo can we split it later per SIG? Or do the opposite and have one discussions per SIG and then combine into one?

**Reply by austinlparker** [reply]
I think it'd be easier to split it up later rather than merge, although I don't know if you can move discussions between repos?

**Reply by tigrannajaryan** [reply]
There is a "Transfer discussion" feature but it is for individual discussions. I don't see a way to do a bulk transfer so it is likely going to be not practical.

**Reply by dyladan** [reply]
> I think it'd be easier to split it up later rather than merge, although I don't know if you can move discussions between repos?

![image](https://user-images.githubusercontent.com/1612643/101672516-f4303b00-3a23-11eb-940f-27f3577abb50.png)

**Reply by austinlparker** [reply]
> > I think it'd be easier to split it up later rather than merge, although I don't know if you can move discussions between repos?
> 
> ![image](https://user-images.githubusercontent.com/1612643/101672516-f4303b00-3a23-11eb-940f-27f3577abb50.png)

Oh, wait, I do see that.  D'oh.

**Comment by SergeyKanzhelev** [comment]
We can consider using categories for this. Discoverability-wise, it will be easier to have them per-language though. Should we have a rule to enable discussions when maintainers of a language request one?

**Reply by austinlparker** [reply]
Yeah, should probably have a category per language and a collector one as well.

**Reply by SergeyKanzhelev** [reply]
Sorry, I meant for discoverability it may be easier to have them in the language repositories. But categories is also ok.

**Comment by dyladan** [comment]
One additional argument for per-repo discussions. Most approvers/contributors most likely only watch the repos they participate in. This watchers of this repo for instance have very little overlap outside of gc/tc with the watchers of the JS repo. As the JS maintainer if I want to solicit the opinions of regular contributors, I wouldn't want to do it in a central repo they may not be watching so I would probably just create an issue.

**Reply by dyladan** [reply]
Also, when most people want to ask a question about a repo they will go to that repo. If there is no discussion tab there, they will most likely open an issue rather than trying to find some centralized community repo with shared discussions.

**Comment by alolita** [comment]
I support having both options as @SergeyKanzhelev suggested earlier on this thread -
1. Language or component specific discussions: If folks have a language specific question or component specific discussion that would tend to happen on the repo itself.
2. Project wide discussions: These could be tagged or in the community repo.

Why not start with the community repo for project wide discussions and then as language or component maintainers need we can set those up.
